### PR TITLE
Header layout

### PR DESCRIPTION
**Pull request by rodrigo-morais**
Adjusting the header's style and layout to issue #1.

Not close the issue because I'll create more PRs to the rest of blog for it.

### CODE REVIEW

**Comment by rodrigo-morais** [comment]
ping @viniciusffj

File: src/style.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+body {
+    background-color: #CCFF0C;
```
Comment (viniciusffj) [inline src/style.css]: Given that we a `MainCSS.elm` that we add some style to body, is this file necessary? 
Can't we move the `background-color: #CCFF0C`  over there?

File: src/assets/header.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.headerNavBar {
```
Comment (viniciusffj) [inline src/assets/header.css]: Since the files `src/assets/styles.css`, `src/assets/header.css` and `src/assets/navigation.css` are generated, we don't need to add them to the repo, right? 

In the [elm-css example](https://github.com/rtfeldman/elm-css/tree/master/examples) they don't add them.

File: src/Navigation/View.elm [hunk 3]
```diff
@@ -1,30 +1,67 @@
...
+        _ ->
+          a [ id viewId
+            , href "javascript://"
+            , onClick message
+            , class'
+            ]
+            [ text label ]
+
+
+    class' =
```
Comment (viniciusffj) [inline src/Navigation/View.elm]: We're repeating the same condition for a really small code. Can't we move the them up? Like:

``` Elm
case isSelected of
  True -> 
    span [ id viewId
         , class [NavigationCss.NavItem, NavigationCss.NavItemSelected]
         ]
         [ text label ]
  _ ->
    a [ id viewId
      , href "javascript://"
      , onClick message
      , class[NavigationCss.NavItem]
      ]
      [ text label ]
```

Comment (viniciusffj) [inline src/Navigation/View.elm]: We're repeating the same condition to a realy small chunk of code. Can't we move the class' logic up? Like this"

``` Elm
case isSelected of
  True -> 
    span [ id viewId
         , class [NavigationCss.NavItem, NavigationCss.NavItemSelected]
         ]
         [ text label ]
  _ ->
    a [ id viewId
      , href "javascript://"
      , onClick message
      , class[NavigationCss.NavItem]
      ]
      [ text label ]
```

File: src/index.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
 'use strict';
 
 require('./assets/markdown.css');
```
Comment (viniciusffj) [inline src/index.js:3]: The files `src/assets/styles.css`, `src/assets/header.css` and `src/assets/navigation.css` are generate by elm-css, why do we need to add them on git?

In the [lib's example](https://github.com/rtfeldman/elm-css/tree/master/examples) they don't add the generated files to git.

File: src/style.css (hunk 1 above)
Comment (viniciusffj) [inline src/style.css]: Since we are applying style to `body` in `MainCss.elm`, can't we move this style to it and delete the `src/style.css`?

**Review by viniciusffj — COMMENTED** [review]
Since we're only adding css to application, why we need the Header module?

IMHO, the `Header` is not necessary, since all the actions goes to `Navigation` module. The only thing it adds is the title mark up and the Header CSS. They could be files leaving in the same module.

Maybe we need to catch up and think a little bit more about this.

File: src/Header/Update.elm [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import Header.Messages exposing (..)
+
+
+import Navigation.Update
+
+
+update : Msg -> Cmd Msg
+update msg =
+  case msg of
+    NoOp ->
```
Comment (viniciusffj) [inline src/Header/Update.elm]: Is this NoOp ever used?

**Comment by viniciusffj** [comment]
@rodrigo-morais I created a branch called `layout-vini` to explain the header part. You can see the changes [here](https://github.com/in-development/in-development-blog/compare/layout...layout-vini?expand=1).

File: src/style.css (hunk 1 above)
Comment (rodrigo-morais) [reply src/style.css]: It was not necessary. Removed!
Thanks!

Comment (rodrigo-morais) [reply src/style.css]: The file was removed.
It was not necessary.

**Comment by rodrigo-morais** [comment]
I joined Header and Navigation.

**Comment by viniciusffj** [comment]
@rodrigo-morais Looks better.

I just have a couple of comments before I merge:
- The files `src/assets/styles.css`, `src/assets/header.css` and `src/assets/navigation.css` are generated by `elm-css`, can't we remove them from git?
- We need to add `node_modules/elm-css/elm-css.js src/Stylesheets.elm` to `npm run build` task,  because we don't use `npm run dev` in prod.

File: src/Header/Update.elm (hunk 5 above)
Comment (rodrigo-morais) [reply src/Header/Update.elm]: Removed.

File: src/assets/header.css (hunk 2 above)
Comment (rodrigo-morais) [reply src/assets/header.css]: Removed.

File: src/Navigation/View.elm (hunk 3 above)
Comment (rodrigo-morais) [reply src/Navigation/View.elm]: I removed the repetition.

**Comment by rodrigo-morais** [comment]
@viniciusffj I didn't finish. Now I removed the files and updated the `build` task.
Could you take a look again?

**Comment by viniciusffj** [comment]
@rodrigo-morais The build is red. I think you misspelled the module `MainCSS` in `Stylesheets.elm` to `MainCss`.

**Comment by rodrigo-morais** [comment]
@viniciusffj Ya. Solved!
Problem with lowercase and uppercase in Linux.

**Comment by viniciusffj** [comment]
@rodrigo-morais That's why we use docker, it helps avoid such things.

We have 18 commits, with duplicated messages, wants squash them so I can merge the pr?

**Comment by rodrigo-morais** [comment]
@viniciusffj What are the duplicated commits?

**Comment by viniciusffj** [comment]
I don't know why, github showed three times the commit "Adding elm-css to "build" task."
Maybe you forced the push and it went crazy.

I'll merge with 18 commtis
